src/Configuration/Loader.php

```diff
@@ -18,8 +20,12 @@ public function load(string $file): Configuration
             throw new MissingFileException();
         }
 
-        return Configuration::fromArray(
-            Yaml::parse(file_get_contents($file))
-        );
+        try {
+            $data = Yaml::parse(file_get_contents($file));
+        } catch (ParseException $exception) {
+            throw FileCannotBeParsedAsYamlException::fromFilename($file);
+        }
+
+        return Configuration::fromArray($data);
```

Comment:
At this point, `$data` could not be an `array`, but I figured I would follow up in a separate PR.